_sub/security/aws-backup-for-rds/main.tf

```diff
@@ -0,0 +1,84 @@
+# AWS Vault Creation
+resource "aws_backup_vault" "vault" {
+  count                     = var.deploy && var.vault_name != "" ? 1 : 0
+  name                      = var.vault_name
+  kms_key_arn               = var.deploy_kms_key ? aws_kms_key.backup[0].arn : var.kms_key_arn
+  tags                      = var.tags
+}
+
+# KMS Key for Encryption
+resource "aws_kms_key" "backup" {
+  count                = var.deploy && var.deploy_kms_key ? 1 : 0
+  description          = "KMS key for backup encryption"
+  enable_key_rotation  = true
+  policy               = data.aws_iam_policy_document.backup_key_policy.json
+  tags                 = var.tags
+}
+
+data "aws_iam_policy_document" "backup_key_policy" {
+  statement {
+    sid       = "Enable IAM User Permissions"
+    actions   = ["kms:*"]
+    resources = ["*"]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"]
+    }
+  }
```

Comment:
I think this statement is very permissive, you allow every user/role in the account to do everything with the key.